src/Receiver/HmacSignature.php

```diff
@@ -0,0 +1,163 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Webhook Module for PHP
+ *
+ * Copyright (c) 2021 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Webhook\Receiver;
+
+use Adyen\AdyenException;
+use Adyen\Webhook\Exception\InvalidDataException;
+
+class HmacSignature
+{
+    const EVENT_CODE = "eventCode";
+    /**
+     * @param string $hmacKey Can be found in Customer Area
+     * @param array $params The response from Adyen
+     * @return string
+     * @throws InvalidDataException
+     */
+    public function calculateNotificationHMAC($hmacKey, $params)
+    {
+        // validate if hmacKey is provided
+        if (empty($hmacKey)) {
+            throw new InvalidDataException("You did not provide a HMAC key");
```

Comment:
Shouldn't this be an HMACKeyValidationException instead?
Do we want to provide such detailed error message? (I'd probably provide back the detailed error message when the mode is test and only the default message back when it's live)

Reply:
We don't have access to live/test mode from this module and to be honest I would prefer to have the plugin/higher-level code be the one to filter the message based on the environment.

Reply:
Agree.
In this case maybe this piece also needs to be adjusted: https://github.com/Adyen/php-webhook-module/pull/1/files#diff-c30e75941de1f56c3bd528314f7e8cadbc7fd90f02d7773eed234c40f1bb363aR59

Reply:
Actually [`isTestNotification()`](https://github.com/Adyen/php-webhook-module/pull/1/files#diff-c30e75941de1f56c3bd528314f7e8cadbc7fd90f02d7773eed234c40f1bb363aR139) only checks the pspReference for the word `test', it does not deal with the environment but test in terms of a test notification sent from CA. I think this could stay? But I want to know what you think

Reply:
yes indeed, let's keep it as is.
Shall we think about the usage of this test flag anywhere else?